front/lib/swr/editors.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from "react";
+import type { Fetcher } from "swr";
+
+import { fetcher, useSWRWithDefaults } from "@app/lib/swr/swr";
+import type { GetAgentEditorsResponseBody } from "@app/pages/api/w/[wId]/assistant/agent_configurations/[aId]/editors";
+import type { LightWorkspaceType } from "@app/types";
+
+export function useEditors({
+  owner,
+  agentConfigurationId,
+  disabled,
+}: {
+  owner: LightWorkspaceType;
+  agentConfigurationId: string | null;
+  disabled?: boolean;
+}) {
+  const editorsFetcher: Fetcher<GetAgentEditorsResponseBody> = fetcher;
+
+  const { data, error, mutate } = useSWRWithDefaults(
+    agentConfigurationId
+      ? `/api/w/${owner.sId}/assistant/agent_configurations/${agentConfigurationId}/editors`
+      : null,
+
+    editorsFetcher,
+    {
+      disabled,
+    }
+  );
+
+  return {
+    editors: useMemo(() => (data ? data.editors : undefined), [data]),
```

Comment:
we usually return an empty array in our swr hooks, as it's easier to handle - any reason to return undefined here ?

btw if you return undefined, the useMemo() is not needed (it's only here to avoid rebuilding empty array - ooc we could replace all of them by an empty array constant)